### PR TITLE
Fix exception message when validating the name of tokenizer(edgeNGram)

### PR DESCRIPTION
**Pull request by YeonghyeonKO**
The method `getTokenizers` in `CommonAnalysisPlugin` returns IllegalArgumentException with the message:
> The [edgeNGram] tokenizer name was deprecated in 7.6. Please use the tokenizer name to [edge_nGram] for indices created in versions 8 or higher instead.

After IndexVersion v.8.0.0, `edge_ngram` (not `edge_nGram`) is the exact name of edgeNGram tokenizer so I ask to change the word in this Pull Request. The PR could reduce confusion in defining edgeNGram tokenizers when statically mapping indices in Elasticsearch.

### CODE REVIEW

**Comment by YeonghyeonKO** [comment]
@mark-vieira Hi, 
- I'd like to ask you that since elastic decided to add **APGL 3.0** to existing licenses, will it be applied starting from v9.0.0? 
(ref : https://www.elastic.co/kr/blog/elasticsearch-is-open-source-again)
- Also, this PR is about index mappings when using **edgeNGram** tokenizer then which team is going to review this PR? (If a meeting for triage is held regularlym, please let me know the schedule.

**Comment by mark-vieira** [comment]
> * I'd like to ask you that since elastic decided to add **APGL 3.0** to existing licenses, will it be applied starting from v9.0.0?
>       (ref : https://www.elastic.co/kr/blog/elasticsearch-is-open-source-again)

As mentioned in the [licensing FAQ](https://www.elastic.co/pricing/faq/licensing#versions) the AGPL change will apply starting with version 8.16.

**Comment by carlosdelest** [comment]
@elasticsearchmachine run elasticsearch-ci

**Comment by carlosdelest** [comment]
@elasticsearchmachine update branch

**Comment by carlosdelest** [comment]
@YeonghyeonKO I tried to update the branch and resolve conflicts, but some compilation errors resulted from it. Can you please update from the branch and correct them?

Thanks!

**Comment by benwtrent** [comment]
@carlosdelest @YeonghyeonKO take a look here: https://github.com/elastic/elasticsearch/pull/113009

**Comment by YeonghyeonKO** [comment]
@carlosdelest Thanks for pulling codes from main branch to this branch. I fixed compilation error by importing classes and defining static variable(deprecationLogger).

However as @benwtrent commented above that after @javanna's PR( https://github.com/elastic/elasticsearch/pull/113009 ) had been merged already, this PR can be CLOSED.

File: modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java [hunk 1]
```diff
@@ -296,6 +303,22 @@ public Map<String, AnalysisProvider<TokenizerFactory>> getTokenizers() {
         tokenizers.put("thai", ThaiTokenizerFactory::new);
         tokenizers.put("ngram", NGramTokenizerFactory::new);
         tokenizers.put("edge_ngram", EdgeNGramTokenizerFactory::new);
+        tokenizers.put("edgeNGram", (IndexSettings indexSettings, Environment environment, String name, Settings settings) -> {
```
Comment (benwtrent) [inline modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java:306]: yeah, this should just be removed :). I think this pr can be closed.

Comment (YeonghyeonKO) [reply modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java:306]: Ok, thanks for reviewing my PR :)

**Comment by javanna** [comment]
This fix was still valid for 8.x I believe, my commit with the removal went only to main.

**Comment by carlosdelest** [comment]
Thanks for clarifying @javanna !

@YeonghyeonKO , can you open another PR with this code, but to [8.x branch](https://github.com/elastic/elasticsearch/tree/8.x), and assign it to me?

Thanks!

**Comment by YeonghyeonKO** [comment]
@carlosdelest 
OK, I'll open a new PR for 8.x version.

**Pull request closed without merging**